checks/fileparser/github_workflow.go

```diff
@@ -208,8 +208,15 @@ func GetOSesForJob(job *actionlint.Job) ([]string, error) {
 	}
 
 	if len(jobOSes) == 0 {
-		return jobOSes, sce.WithMessage(sce.ErrScorecardInternal,
-			fmt.Sprintf("unable to determine OS for job: %v", GetJobName(job)))
+		// This error is caught by the caller, which is responsible for adding more
+		// precise location information
```

Comment:
optional: If the caller needs to add more information anyway, can we just return `sce.ErrJobOSParsing` here and let the caller create the `ElementError`?

It seems a little brittle to have 2 functions composing the error when the jobName is available to the caller too.